jwt-authorizer/src/layer.rs

```diff
@@ -209,29 +266,35 @@ where
     type Future = BoxFuture<'static, Result<Request<B>, AuthError>>;
 
     fn authorize(&self, mut request: Request<B>) -> Self::Future {
-        let authorizer = self.auth.clone();
-        let h = request.headers();
+        let tkns_auths: Vec<(Option<String>, Arc<Authorizer<C>>)> = self
+            .auths
+            .iter()
+            .map(|a| (a.extract_token(request.headers()), a.clone()))
+            .collect();
+
+        if !tkns_auths.iter().any(|(t, _)| t.is_some()) {
+            return Box::pin(future::ready(Err(AuthError::MissingToken())));
+        }
 
-        let token = match &self.jwt_source {
-            layer::JwtSource::AuthorizationHeader => {
-                let bearer_o: Option<Authorization<Bearer>> = h.typed_get();
-                bearer_o.map(|b| String::from(b.0.token()))
-            }
-            layer::JwtSource::Cookie(name) => h
-                .typed_get::<headers::Cookie>()
-                .and_then(|c| c.get(name.as_str()).map(String::from)),
-        };
         Box::pin(async move {
-            if let Some(token) = token {
-                authorizer.check_auth(token.as_str()).await.map(|token_data| {
+            let mut token_data: Result<TokenData<C>, AuthError> = Err(AuthError::NoAuthorizer());
+            for (tkn, auth) in tkns_auths {
+                if let Some(token) = tkn {
+                    token_data = auth.check_auth(token.as_str()).await;
```

Comment:
This ends up doing the checks sequentially rather then potentially in parallel; Which won't matter much if the checks don't end up doing network calls ofcourse, but they might.  So doing a FuturesUnordered as i did in #22  may be beneficial

Reply:
Depends ... on weather we want to optimise the number of external calls or the time passed in authorizer or you want order the calls calling some authorizers as a fallback ... but I think you are right but not sure.